aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -67,23 +80,40 @@ sub get_anaconda_version
     return $version;
 }
 
-# Returns the absolute path where the PXE file must be written.
+# Prints an IP address in hexadecimal.
+sub hexip
+{
+    my ($ip, $variant) = @_;
+
+    my $hexip_str='';
+    if ( $variant == PXE_VARIANT_PXELINUX ) {
+        hexip_str = sprintf ("%02X%02X%02X%02X", $1, $2, $3, $4) if ($ip =~ m/^(\d+)\.(\d+)\.(\d+)\.(\d+)$/);
```

Comment:
please reuse the comon sprintf part; it's very hard to see if these formats are the same or not